### ISSUE TITLE
Why not use cross-entropy loss?

### ISSUE THREAD

**Issue by TwoWhite2** [rhythmcao/text2sql-lgesql#8]
Why not use the cross-entropy loss function but only maximize the probability of a positive category？

**Comment by rhythmcao** [comment]
I suppose these two statements are the same.
If you mean why we do not directly use the CrossEntropyLoss API from pytorch for the main text-to-SQL loss, type constaints are incorporated in the decoder, such that the probability for ApplyRule/SelectTable/SelectColumn are softmaxed separately. They can be concatenated together for more efficient implementation, but directly extract the probability of a positive category is ease of use and more readable.
